frontends/p4/fromv1.0/converters.cpp

```diff
@@ -43,12 +43,15 @@ const IR::Node* ExpressionConverter::postorder(IR::Mask* expression) {
     auto exp = expression->left;
     auto cst = expression->right->to<IR::Constant>();
     mpz_class value = cst->value;
-    while (value != 0) {
-        auto range = Util::findOnes(value);
-        value = value - range.value;
+    auto range = Util::findOnes(value);
+    if (value != range.value) {
+        /* FIXME -- can't represent the mask as a single slice -- no way to express this
```

Comment:
The key can be expressed as a concatenation of slices.
Alternatively, the key could be a slice from the first 1 to the last 1 bit.
The actual representation depends on how the information is used in the back-end and in the control-plane API. Until we have a specification on how that is done we cannot really write this code.